docs/source/data/fetching-rest.mdx

```diff
@@ -1,77 +1,114 @@
+  }
 
-```ts {4-9} title="index.js"
-const server = new ApolloServer({
+  async getMostViewedMovies(limit = '10'):Promise<Movie[]> {
+    const data = await this.get('movies', {
+      params: {
+        per_page: limit,
+        order_by: 'most_viewed',
+      },
+    });
+    return data.results;
+  }
+}
+```
+
+</MultiCodeBlock>
+
+You can extend the `RESTDataSource` class to implement whatever data-fetching methods your resolvers need. These methods can use built-in convenience methods (like `get` and `post`) to perform HTTP requests, helping you add query parameters, parse JSON results, and handle errors.
+
+## Adding subclasses to Apollo Server's context
+
+You can then provide your `RESTDataSource` subclasses to the `context` initialization function, like so:
+
+<MultiCodeBlock>
+
+```ts title="index.ts"
+//highlight-start
+interface ContextValue {
+  dataSources: {
+    moviesAPI: MoviesAPI;
+    personalizationAPI: PersonalizationAPI;
+  };
+}
+//highlight-end
+
+const server = new ApolloServer<ContextValue>({
   typeDefs,
   resolvers,
-  csrfPrevention: true,
-  cache: 'bounded',
-  plugins: [ApolloServerPluginLandingPageLocalDefault({ embed: true })],
-  dataSources: () => {
+});
+
+const { url } = await startStandaloneServer(server, {
+  context: async ({ req }) => {
     return {
-      moviesAPI: new MoviesAPI(),
-      personalizationAPI: new PersonalizationAPI(),
+      //highlight-start
+      // We create new instances of our data sources with each request
+      dataSources: {
+        moviesAPI: new MoviesAPI(),
+        personalizationAPI: new PersonalizationAPI(),
+      },
+      //highlight-end
     };
   },
 });
+
+console.log(`🚀  Server ready at ${url}`);
 ```
 
-- As shown, the `dataSources` option is a _function_. This function returns an _object_ containing instances of your `DataSource` subclasses (in this case, `MoviesAPI` and `PersonalizationAPI`).
-- Apollo Server calls this function for _every incoming operation_. It automatically assigns the returned object to the `dataSources` field of [the `context` object](./resolvers/#the-context-argument) that's passed between your server's resolvers.
-- Also as shown, **the function should create a new instance of each data source for each operation.** If multiple operations share a single data source instance, you might accidentally combine results from multiple operations.
+</MultiCodeBlock>
+
+<!-- TODO(AS4) add link to context article once exists  -->
+Apollo Server calls the [the `context` initialization](../resolvers/#the-context-argument) function for _every incoming operation_. This means:
+- As shown above, with every operation `context` returns an _object_ containing new instances of your `RESTDataSource` subclasses (in this case, `MoviesAPI` and `PersonalizationAPI`).
+-  These subclasses are accessible from the [the `context` argument](./resolvers/#the-context-argument) that's passed between your server's resolvers.
```

Comment:
Are we calling this `contextValue` everywhere or no? We might need to rename the heading, on the resolvers doc, etc.

Reply:
Yeah, I'm going to do that in a separate PR after merging in all the AS4 docs in, so I can change `contextValue` everywhere all at once. I think it'll be less error-prone that way  😅